cmd/migration/main_test.go

```
@@ -0,0 +1,286 @@
+func buildMigrationBinary(t *testing.T) string {
+	t.Helper()
+	migrationBinOnce.Do(func() {
+		dir, err := os.MkdirTemp("", "migration-bin")
+		if err != nil {
+			migrationBinErr = err
+			return
+		}
+		bin := filepath.Join(dir, "migration")
+		// Build the current package (cmd/migration). The test's working
+		// directory is the package directory.
+		cmd := exec.Command("go", "build", "-mod=mod", "-o", bin, ".")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)